lib/npg_pipeline/daemon.pm

```diff
@@ -172,43 +108,10 @@ sub staging_host_match {
 sub check_lims_link {
   my ($self, $run) = @_;
 
-  my $fc_barcode = $run->flowcell_id;
-  if(!$fc_barcode) {
-    croak q{No flowcell barcode};
-  }
-
   my $batch_id = $run->batch_id();
-  my $ref = { 'iseq_flowcell'  => $self->iseq_flowcell };
-  $ref->{'flowcell_barcode'} = $fc_barcode;
-  $ref->{'id_flowcell_lims'} = $batch_id;
+  $batch_id or $self->logcroak(q{No batch id});
 
-  my $obj = $self->lims_query_class()->new_object($ref);
-  my @fcell_rows = $obj->query_resultset()->all();
-  my $fcell_row = $fcell_rows[0];
-
-  if ( !($batch_id || $fcell_row)  ) {
-    croak q{No matching flowcell LIMs record is found};
-  }
-
-  my $lims = {};
-  $lims->{'id'} = $batch_id;
-  if ($fcell_row) {
-    $lims->{'qc_run'} = (defined $fcell_row->purpose && $fcell_row->purpose eq 'qc') ? 1 : undef;
-  } else {
-    $lims->{'qc_run'} = npg_pipeline::base::options->is_qc_run($lims->{'id'});
-    if (!$lims->{'qc_run'}) {
-      croak q{Not QC run and not in the ml warehouse};
-    }
-  }
-
-  my @studies = ();
-  if (!$lims->{'qc_run'}) {
-    @studies = uniq map { $_->study_id } grep { !$_->is_control } @fcell_rows;
-    @studies = sort @studies;
-  }
-  $lims->{'studies'} = \@studies;
-
-  return $lims;
+  return {id => $batch_id};
```

Comment:
I was thinking this method can be changed to `get_batch_id`. Then just return the id or logcroak. But the analysis daemon child depends on a hash to be returned. So it will need to be refactored there too with a few lines changed. So not sure how you feel about the whole chain of changes.